packages/phone-number-privacy/test/end-to-end/get-blinded-salt.test.ts

```diff
@@ -0,0 +1,108 @@
+import { serializeSignature, signMessage } from '@celo/utils/lib/signatureUtils'
+import 'isomorphic-fetch'
+import { getBlindedPhoneNumber } from '../utils'
+
+require('dotenv').config()
+
+const PHONE_NUM_PRIVACY_SERVICE = process.env.PHONE_NUM_PRIVACY_SERVICE_URL
+const SIGN_MESSAGE_ENDPOINT = '/getBlindedSalt'
+
+const PRIVATE_KEY1 = '0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef'
+const ACCOUNT_ADDRESS1 = normalizeAddressWith0x(privateKeyToAddress(PRIVATE_KEY1))
+const PRIVATE_KEY2 = '0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890fdeccc'
+const ACCOUNT_ADDRESS2 = privateKeyToAddress(PRIVATE_KEY2)
+const PHONE_NUMBER = '+15555555555'
+const IDENTIFIER = PhoneNumberUtils.getPhoneHash(PHONE_NUMBER)
+const BLINDING_FACTOR = new Buffer('0IsBvRfkBrkKCIW6HV0/T1zrzjQSe8wRyU3PKojCnww=', 'base64')
+const BLINDED_PHONE_NUMBER = getBlindedPhoneNumber(PHONE_NUMBER, BLINDING_FACTOR)
+
+describe('Running against a deployed service', () => {
+  describe('Returns status 400 with invalid input', () => {
+    it('With invalid address', async () => {
+      const response = await postToSignMessage(
+        BLINDED_PHONE_NUMBER,
+        PRIVATE_KEY1,
+        '0x1234',
+        'ignore'
+      )
+      expect(response.status).toBe(400)
+    })
+
+    it('With missing blindedQueryPhoneNumber', async () => {
+      const response = await postToSignMessage('', PRIVATE_KEY1, ACCOUNT_ADDRESS1)
+      expect(response.status).toBe(400)
+    })
+  })
+
+  describe('Returns status 401 with invalid authentication headers', () => {
+    it('With invalid auth header', async () => {
+      const response = await postToSignMessage(
+        BLINDED_PHONE_NUMBER,
+        PRIVATE_KEY1,
+        ACCOUNT_ADDRESS1,
+        'invalid'
+      )
+      expect(response.status).toBe(401)
+    })
+
+    it('With auth header signer mismatch', async () => {
+      // Sign body with different account
+      const body = JSON.stringify({
+        hashedPhoneNumber: '+1455556600',
+        blindedQueryPhoneNumber: BLINDED_PHONE_NUMBER.trim(),
+        ACCOUNT_ADDRESS1,
+      })
+      const signature = signMessage(JSON.stringify(body), PRIVATE_KEY2, ACCOUNT_ADDRESS2)
+      const authHeader = serializeSignature(signature)
+
+      const response = await postToSignMessage(
+        BLINDED_PHONE_NUMBER,
+        PRIVATE_KEY1,
+        ACCOUNT_ADDRESS1,
+        authHeader
+      )
+      expect(response.status).toBe(401)
+    })
+
+    it('With missing blindedQueryPhoneNumber', async () => {
+      const response = await postToSignMessage('', PRIVATE_KEY1, ACCOUNT_ADDRESS1)
+      expect(response.status).toBe(400)
+    })
+  })
+
+  it('Address salt querying out of quota', async () => {
+    const response = await postToSignMessage(BLINDED_PHONE_NUMBER, PRIVATE_KEY1, ACCOUNT_ADDRESS1)
+    expect(response.status).toBe(403)
+  })
+})
+
+async function postToSignMessage(
```

Comment:
so does running this require we have the service running locally?

Reply:
Yes, there must be a service running for these tests. The tester can run it locally and just run the tests, or point the .env parameter to a deployed instance. This is an easy way to perform validation after you've deployed an upgrade to staging for example. I want this to eventually be automated as part of our CICD process.